screenshots.rst

```diff
@@ -0,0 +1,263 @@
+Automating Form-Widgets Guide
+===================================
+
+This guide gives information on how to automate generation of screenshots for form-widgets guide as well as regenerating form-widgets guide whenever new widgets are added.
+
+.. _screenshot-screengrab:
+
+Screenshot Generation using Screengrab
+----------------------------------------
+
+`Screengrab <https://docs.fastlane.tools/actions/screengrab/>`_ is a command line tool, provided by fastlane. fastlane is an open source platform, which provides a toolchain and automates tasks like generating screenshots, dealing with code signing, and releasing your application. Screengrab can be configured to load up different locales, run all the UI tests and take screenshots during the execution of the test on various devices. 
+
+.. _setting-screengrab:
+
+Setting up Screengrab
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To configure screengrab, follow the steps given below:
+
+.. _install-screen:
+
+Installation
+""""""""""""""""""
+
+In the command line type, the following command to check if RubyGem is installed or not. RubyGems is a package manager for the Ruby programming language that provides a standard format for distributing Ruby programs and libraries (in a self-contained format called a *gem*). If Ruby is installed, the command given below would return the version number.
+
+.. code-block:: console
+
+  $ gem -v
+
+If it is not installed, you can install it using the instructions `given here <https://rubygems.org/pages/download>`_. Windows users are recommended to use `RubyInstallers <https://rubyinstaller.org/>`_.
+
+After installing RubyGem install the gems:
+
+Windows and Mac users can use the following commands:
```

Comment:
On a stock macOS 10.13.3 running ruby 2.3, I get this error

```
mkmf.rb can't find header files for ruby at /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/include/ruby.h
```

I fixed this by running `xcode-select --install`. 

Then I got another a permission error.
```
You don't have write permissions for the /Library/Ruby/Gems/2.3.0 directory
```

I fixed that by `sudo gem install fastlane`

This is roughly inline with what the fastlane docs say at https://docs.fastlane.tools/getting-started/ios/setup/.


@adammichaelwood Is this the error you got?

Reply:
No. I got some other errors regarding the Ruby version (I was on 2.0, not 2.3).
Something, somewhere said I needed to be on 2.4, and several attempts to upgrade failed.

However --- I have successfully used Homebrew to upgrade to 2.3, and am trying your procedure above.....

Reply:
I am now sorted. gems installed. i will run the screenshot script early tomorrow morning and report back